### PR TITLE
Second fix for tx by hash in evm fork

### PR DESCRIPTION
**Pull request by vldm**

### CODE REVIEW

File: ledger/src/blockstore.rs [hunk 1]
```diff
@@ -2561,22 +2565,35 @@ impl Blockstore {
...
+            for primary_index in 0..=1 {
+                if let Some(tx) = self
+                    .evm_transactions_cf
+                    .get_protobuf_or_bincode::<evm::TransactionReceipt>((
+                        primary_index,
+                        signature,
+                        block_num,
+                        maybe_slot,
+                    ))?
+                    .and_then(|meta| meta.try_into().ok())
```
Comment (hrls) [inline ledger/src/blockstore.rs:2593]: What if all values falls into Err variant?

File: ledger/src/blockstore.rs [hunk 2]
```diff
@@ -2642,36 +2663,55 @@ impl Blockstore {
...
-                continue;
-            };
-            let confirmed_block_found = blocks
-                .take_while(|((block_index, _slot), _block)| block_index == block_num)
-                .any(|(_, header)| {
-                    self.is_root(header.native_chain_slot)
-                        && header.transactions.contains(found_hash)
-                });
+        // find first transaction with confirmed slot
+        for ((_p, _found_hash, _block_num, slot), data) in &transactions {
```
Comment (hrls) [inline ledger/src/blockstore.rs]: `Iterator:find`

File: ledger/src/blockstore.rs [hunk 3]
```diff
@@ -2705,8 +2746,10 @@ impl Blockstore {
         let primary_index =
             self.get_primary_index(block_number, &mut w_active_transaction_status_index)?;
         let status = status.into();
-        self.evm_transactions_cf
-            .put_protobuf((primary_index, hash, block_number), &status)?;
+        self.evm_transactions_cf.put_protobuf(
+            (primary_index, hash, block_number, Some(slot_index)),
```
Comment (hrls) [inline ledger/src/blockstore.rs]: New type

File: ledger/src/blockstore_db.rs [hunk 4]
```diff
@@ -832,33 +835,46 @@ impl ProtobufColumn for columns::EvmHeaderIndexByHash {
 }
 
 impl Column for columns::EvmTransactionReceipts {
-    type Index = (u64, H256, Slot);
-
-    fn key((index, hash, slot): (u64, H256, Slot)) -> Vec<u8> {
-        let mut key = vec![0; 8 + 32 + 8]; // size_of u64 + size_of HASH + size_of Slot
+    type Index = (u64, H256, evm_state::BlockNum, Option<Slot>);
```
Comment (hrls) [inline ledger/src/blockstore_db.rs]: New type

**Review by hrls — APPROVED** [review]

File: ledger/src/blockstore.rs (hunk 1 above)
Comment (vldm) [reply ledger/src/blockstore.rs:2593]: Then will return Empty list.

**Pull request merged**